cpp/dolfinx/fem/FiniteElement.cpp

```diff
@@ -403,8 +403,25 @@ FiniteElement::get_dof_permutation_function(bool inverse,
     }
     else if (!scalar_element)
     {
-      throw std::runtime_error(
-          "Permuting DOFs for vector elements not implemented.");
+      // Vector element
+      std::function<void(const xtl::span<std::int32_t>&, std::uint32_t)>
+          sub_element_function
+          = _sub_elements[0]->get_dof_permutation_function(inverse);
+      int dim = _sub_elements[0]->space_dimension();
+      int bs = _bs;
+      return [dim, sub_element_function,
+              bs](const xtl::span<std::int32_t>& doflist,
+                  std::uint32_t cell_permutation) {
+        std::vector<std::int32_t> subdofs(dim);
```

Comment:
Does this give a dynamic memory allocation every time the function is called?

Reply:
Yes, but it's difficult to get rid of unless we get Basix to handle block size when doing permutations

Reply:
How about 'capturing' the vector like we do for the lambda that holds PETSc matrices?